Classification/include/CGAL/Classification/Mesh_feature_generator.h

```diff
@@ -149,31 +151,30 @@ class Mesh_feature_generator
 
   struct Scale
   {
-    Neighborhood* neighborhood;
-    Planimetric_grid* grid;
-    Local_eigen_analysis* eigen;
+    std::unique_ptr<Neighborhood> neighborhood;
+    std::unique_ptr<Planimetric_grid> grid;
+    std::unique_ptr<Local_eigen_analysis> eigen;
     float voxel_size;
 
     Scale (const FaceListGraph& input,
            const Face_range& range,
            PointMap point_map,
            const Iso_cuboid_3& bbox, float voxel_size,
            std::size_t nb_scale,
-           Planimetric_grid* lower_grid = nullptr)
+           std::unique_ptr<Planimetric_grid> const &lower_grid = nullptr)
```

Comment:
Do not use a reference here. It is a pointer and it is a light object. Reference is actually bad here. Especially, since you assign a `nullptr` to it. Change it everywhere.

Reply:
Here the use of reference is not to save time by avoiding copy. But since, we are passing a std::unique_ptr as argument so copy won't be possible then there are two other option use std::move(ptr_name) or pass it as a const reference. I sought help from:https://stackoverflow.com/questions/30905487/how-can-i-pass-stdunique-ptr-into-a-function.

Reply:
My bad! I forgot about that! Great. Then nothing to fix.